android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/survey/SurveyScreen.kt

```diff
@@ -35,79 +27,78 @@ fun SurveyScreen(
+            // TODO submit survey
         }
     )
 }
 
+@OptIn(ExperimentalPagerApi::class)
 @Composable
 private fun SurveyScreenContent(
+    questions: List<QuestionUiModel>,
     onBackClick: () -> Unit,
-    onStartClick: () -> Unit
+    onSubmitClick: () -> Unit
 ) {
-    Box(
-        modifier = Modifier.fillMaxSize()
-    ) {
-        DimmedImageBackground(
-            // TODO fetch survey detail https://github.com/luongvo/kmm-survey/issues/23
-            imageUrl = "https://dhdbhh0jsld0o.cloudfront.net/m/1ea51560991bcb7d00d0_l"
-        )
+    val pagerState = rememberPagerState()
+    val scope = rememberCoroutineScope()
 
-        BackButton(
-            modifier = Modifier
-                .statusBarsPadding()
-                .padding(vertical = 5.dp)
-                .semantics { this.contentDescription = SurveyBackButton },
-            onClick = onBackClick
-        )
-
-        Column(
-            modifier = Modifier
-                .statusBarsPadding()
-                .navigationBarsPadding()
-                .fillMaxWidth()
-                .padding(top = 70.dp)
-                .padding(horizontal = dimensions.paddingMedium)
-        ) {
-            Text(
-                // TODO fetch survey detail https://github.com/luongvo/kmm-survey/issues/23
-                text = "Scarlett Bangkok",
-                color = White,
-                style = typography.h4,
-                maxLines = 2,
-                overflow = TextOverflow.Ellipsis
-            )
-            Spacer(modifier = Modifier.height(dimensions.paddingSmall))
-            Text(
-                // TODO fetch survey detail https://github.com/luongvo/kmm-survey/issues/23
-                text = "We'd love to hear from you!",
-                color = White70,
-                style = typography.body1,
-                maxLines = 2,
-                overflow = TextOverflow.Ellipsis
+    HorizontalPager(
+        count = questions.size + 1,
+        state = pagerState,
+        userScrollEnabled = false,
+        modifier = Modifier.fillMaxSize()
+    ) { index ->
+        // TODO use question.displayType instead
+        if (index == 0) {
+            SurveyIntro(
+                onBackClick = onBackClick,
+                onStartClick = { pagerState.scrollToNextPage(scope) }
             )
-            Spacer(modifier = Modifier.weight(1f))
-            PrimaryButton(
-                text = stringResource(id = R.string.survey_start),
-                onClick = onStartClick,
-                modifier = Modifier
-                    .wrapContentWidth()
-                    .align(Alignment.End)
-                    .padding(bottom = dimensions.paddingLargest)
+        } else {
+            SurveyQuestion(
+                index = index,
+                count = questions.size,
+                question = questions[index - 1],
+                onCloseClick = onBackClick,
+                onNextClick = {
+                    if (index != questions.size) {
```

Comment:
@luongvo Optional, could we consider putting `onSubmitClick`? It would be more readable and better encapsulation. 
We don't need more steps to check the condition either (`if (index != questions.size)`).
What do you think?

Reply:
@doannimble I think I can trigger the  `onSubmitClick` from `SurveyQuestion` view, let me try to make an update 👌

Reply:
Is this what you mean https://github.com/luongvo/kmm-survey/pull/68/commits/c649753a28afb71a42a76f8e476afd858e0e322b? @doannimble 😛